### ISSUE TITLE
Installation Lessons: Bring format inline with lesson format

### ISSUE THREAD

**Issue by JoshDevHub** [TheOdinProject/curriculum#27160]
### Description

The installation guides currently disobey the [LAYOUT_STYLE_GUIDE](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md), and this results in them failing against the new Markdown linter that runs on pull request to this repo.

These lessons should be converted over to a format where they can fit the lesson layout. One idea for how to do this:
> I feel like all of these lessons could probably be restructured to follow the typical lesson structure. The installation sections could be restructured as the "assignment" of the lesson, the lesson overview might just end up being "Installing some Cool Thing', and the knowledge check we might be able to throw in some easy things that could be added as an assignment item, like a resource on basics of whatevers being installed.

See also: #27146 

This will probably take a bit of discussion and feeling things out for what's the best idea (feel free to leave comments on this). When we have a good idea in place, I'll begin assigning contributors to work on updating the affected lessons listed below.

### Affected Lessons

- [ ] [Installations](https://www.theodinproject.com/lessons/foundations-installations)
- [ ] [Text Editors](https://www.theodinproject.com/lessons/foundations-text-editors)
- [ ] [Installing Node.js](https://www.theodinproject.com/lessons/foundations-installing-node-js)
- [ ] [Setting up Git](https://www.theodinproject.com/lessons/foundations-setting-up-git)
- [ ] [Installing Ruby](https://www.theodinproject.com/lessons/ruby-installing-ruby)
- [ ] [Installing PostgreSQL](https://www.theodinproject.com/lessons/ruby-on-rails-installing-postgresql)

There are other lessons listed in the [installations guides](https://www.theodinproject.com/guides/installations), but the other ones fail the lesson layout style in different ways. So they're not going to fit cleanly with the approach/solution we settle on here.

**Comment by Churtified** [comment]
I can update a lesson once we decide how to proceed :)

**Comment by wise-king-sullyman** [comment]
@thatblindgeye would you be able to do an a11y review of these lessons as the first step for this issue?

**Comment by mao-sz** [comment]
Just for audit trail, I think the plan we agreed on (which has now been implemented for the OS + Chrome installation lesson) is:

1. Extract OS-specific installation instructions to separate guides, hosted on GH for now.
1. Amend site-hosted lessons to link to our now external guides.
1. Work on the website codebase to be able to host these guides on the website itself (not as lessons in courses).

That should resolve the a11y concerns regarding `<details>`/`<summary>` as well as duplicate headings in the page.

**Comment by zachmmeyer** [comment]
@mao-sz Correct. I went ahead and updated the main ticket, checking off the first checkbox.
